src/broadcast/participant.rs

```diff
@@ -105,13 +119,14 @@ impl ProtocolParticipant for BroadcastParticipant {
     ) -> Result<ProcessOutcome<Self::Output>> {
         info!("Processing broadcast message.");
 
-        // XXX Protocols that utilize `BroadcastParticipant` never "reset" the
-        // participant between executions, and hence an already terminated
-        // `BroadcastParticipant` may get called again!
-        //
-        // if self.is_done() {
-        //     return Err(InternalError::ProtocolAlreadyTerminated);
-        // }
+        if let Status::ParticipantCompletedBroadcast(participants) = self.status() {
+            // The protocol has terminated if the number of participants who
+            // have completed a broadcast equals the total number of other
+            // participants.
+            if participants.len() == self.other_participant_ids.len() {
```

Comment:
Should this be `other + 1`? Because every participant, including this one, needs to finish broadcasting.

Reply:
Oh, that would make sense.

Reply:
Yep it looks like that's the case. We generate our message in `gen_round_one_msgs` but only run the `process_votes` method on `handle_`d messages, so we only vote for other participants, not ourselves.